src/ucp/api/ucp.h

```diff
@@ -1794,6 +1857,49 @@ void ucp_rkey_destroy(ucp_rkey_h rkey);
 
 /**
  * @ingroup UCP_COMM
+ * @brief Non-blocking stream send operation.
+ *
+ * This routine sends a data that is described by the local address @a
+ * buffer, size @a count, and @a datatype object to the destination endpoint
+ * @a ep. The routine is non-blocking and therefore returns immediately, however
+ * the actual send operation may be delayed. The send operation is considered
+ * completed when it is safe to reuse the source @e buffer. If the send
+ * operation is completed immediately the routine return UCS_OK and the call-back
+ * function @a cb is @b not invoked. If the operation is @b not completed
+ * immediately and no error reported then the UCP library will schedule to
+ * invoke the call-back @a cb whenever the send operation will be completed. In
+ * other words, the completion of a operation can be signaled by the return code
+ * or the call-back.
```

Comment:
I suggest re-wording the last two sentences as follows:

If the operation is \@b not completed immediately and no error is reported, then the UCP library will schedule
invocation of the call-back \@a cb upon completion of the send operation. In other words, the completion of the
operation will be signaled either by the return code or by the call-back.